bdai_ros2_wrappers/bdai_ros2_wrappers/service_client.py

```diff
@@ -0,0 +1,21 @@
+# Copyright [2023] Boston Dynamics AI Institute, Inc.
+
+from bdai_ros2_wrappers.node import NodeWrapper
+
+
+class ServiceClientWrapper:
+    """A wrapper around a service client that allows the service to be safely called from callbacks without
+    deadlocking."""
+
+    def __init__(self, node_name: str, service_type, service_name: str):
+        self._node_wrapper = NodeWrapper(node_name)
+        self._service_name = service_name
+        self._client = self._node_wrapper.node.create_client(service_type, service_name)
+
+    def call(self, request, timeout_sec=None):
+        """Calls the service and returns the result. This is safe to call from a callback."""
+        while not self._client.wait_for_service(timeout_sec=timeout_sec):
+            self._node_wrapper.node.get_logger().info(f"{self._service_name} not available, waiting again...")
```

Comment:
Should this potentially loop forever or for a fixed number of times then raise an exception? Also, maybe use warn so it will stand out against all the info messages